starknet_py/net/account/account.py

```diff
@@ -143,7 +146,7 @@ async def _prepare_invoke_function(
 
         max_fee = await self._get_max_fee(transaction, max_fee, auto_estimate)
 
-        return _add_max_fee_to_transaction(transaction, max_fee)
+        return cast(Invoke, _add_max_fee_to_transaction(transaction, max_fee))
```

Comment:
Why do we need a cast here?

Reply:
Because `_add_max_fee_to_transaction` returns the `AccountTransaction` but this method should return `Invoke`. Typechecker is complaining :/

Reply:
I've changed all `Union[Invoke, Declare, DeployAccount]` to the `AccountTransaction` and typechecker started complaining after that